secp256k1-sys/src/lib.rs

```diff
@@ -119,10 +124,19 @@ impl SchnorrSigExtraParams {
     }
 }
 
-/// A Secp256k1 context, containing various precomputed values and such
-/// needed to do elliptic curve computations. If you create one of these
-/// with `secp256k1_context_create` you MUST destroy it with
-/// `secp256k1_context_destroy`, or else you will have a memory leak.
+/// An opaque Secp256k1 context.
+///
+/// Currently this object contains a blinding factor used internally to
+/// randomize computations to protect against sidechannel attacks. In the
+/// past it has contained precomputation tables to speed up crypto operations.
+///
+/// It should be assumed to be expensive to create and therefore should be
+/// reused when possible.
+///
+/// If you create one of these with `secp256k1_context_create` you must
+/// destroy it with `secp256k1_context_destroy`. (Failure to destroy it is
+/// a memory leak; destroying it using any other allocator is likely to be
+/// undefined behavior.)
```

Comment:
I'd just write "is undefined behavior". I don't think it's useful to try to define some crazy circumstances when it's not.

Reply:
Sure.